src/lib/router.ts

```diff
@@ -29,22 +32,24 @@ export function useDefaultNavigation(event: MouseEvent) {
 export const base = import.meta.env.VITE_HASH_ROUTING ? "./" : "/";
 
 // Gets triggered when clicking on an anchor hash tag e.g. <a href="#header"/>
-// Allows the jump to a anchor hash
-window.addEventListener("hashchange", e => {
-  const route = pathToRoute(e.newURL);
+// Allows the jump to a anchor hash.
+window.addEventListener("hashchange", async (event: HashChangeEvent) => {
+  const route = pathToRoute(event.newURL);
   if (route?.resource === "projects" && route.params.hash) {
     if (route.params.hash.match(/^L\d+$/)) {
-      updateProjectRoute({ line: route.params.hash });
+      await updateProjectRoute({ line: route.params.hash });
     } else {
-      updateProjectRoute({ hash: route.params.hash });
+      await updateProjectRoute({ hash: route.params.hash });
     }
   }
 });
 
 // Replaces history on any user interaction with forward and backwards buttons
-// with the current window.history.state
-window.addEventListener("popstate", e => {
-  if (e.state) replace(e.state);
+// with the current window.history.state.
+window.addEventListener("popstate", async (event: PopStateEvent) => {
+  if (event.state) {
+    await replace(event.state);
+  }
```

Comment:
A consequence of the `if` statement is that the back or forward buttons only works if there is a `state` in the history. This should always be the case and this invariant is enforced by the implementations in this module. But we never make this invariant explicit so it is easy to break this if you’re not aware of this when you make changes to this module.

To guard against this I would suggest to add an assertion that `event.state` should always be set. 
```suggestion
  if (!event.state) {
    throw new Error("History entry does not have state");
  }
  await replace(event.state);
```

Alternatively, you could stop relying on state to be set and instead get the route from the location. This should just work:
```typescript
window.addEventListener("popstate", () => initialize());
```

(In this case I’d rename `initialize()` to `loadFromLocation()`.)